core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -162,6 +162,34 @@ package object experimental {
     */
   trait NoChiselNamePrefix
 
+  /** Treat instances of this type as a named component, and rename using the Chisel compiler
+    * plugin
+    *
+    * The compiler plugin by default automatically names instances of `Data` and `Mem`
+    * recursively. This trait allows the plugin to treat arbitrary types as named components
+    * and prefix them when they otherwise would not be
```

Comment:
```suggestion
    * recursively. This trait allows the plugin to treat arbitrary types as named components
    * and prefix them (and objects constructed within them) when they otherwise would not be.
```